lib/component.cpp

```diff
@@ -1,10 +1,118 @@
 #include "component.hpp"
+#include "logical_address.hpp"
+#include "messages/spa_subscription_reply.hpp"
+#include "messages/spa_subscription_request.hpp"
+#include "messages/spa_data.hpp"
+#include "messages/op_codes.hpp"
+#include <iostream>
+#include <memory>
 
-void Component::sendMsg(std::shared_ptr<SpaMessage> message)
+void Component::registerSubscriptionRequest(std::shared_ptr<SpaMessage> message)
 {
-  if (message == nullptr || communicator == nullptr)
+  if (message == nullptr)
   {
+    std::cout << "Invalid subscription request" << std::endl;
+  }
+
+  auto castMessage = std::dynamic_pointer_cast<SpaSubscriptionRequest>(message);
+
+  // Dynamic memory allocation !!!
```

Comment:
This comment is probably not necessary, we are allocating memory in lots of places.